common/server_game.cpp

```diff
@@ -287,18 +287,19 @@ void Server_Game::sendGameStateToPlayers()
     // If spectators are not omniscient, we need an additional createGameStateChangedEvent call, otherwise we can use
     // the data we used for the replay. All spectators are equal, so we don't need to make a createGameStateChangedEvent
     // call for each one.
-    Event_GameStateChanged spectatorEvent;
-    if (spectatorsSeeEverything)
-        spectatorEvent = omniscientEvent;
-    else
-        createGameStateChangedEvent(&spectatorEvent, 0, false, false);
+    Event_GameStateChanged spectatorNormalEvent;
+    createGameStateChangedEvent(&spectatorNormalEvent, nullptr, false, false);
 
     // send game state info to clients according to their role in the game
     for (Server_Player *player : players.values()) {
         GameEventContainer *gec;
-        if (player->getSpectator())
-            gec = prepareGameEvent(spectatorEvent, -1);
-        else {
+        if (player->getSpectator()) {
+            if (spectatorsSeeEverything || player->getJudge()) {
+                gec = prepareGameEvent(omniscientEvent, -1);
+            } else {
+                gec = prepareGameEvent(spectatorNormalEvent, -1);
+            }
+        } else {
             Event_GameStateChanged event;
```

Comment:
I like the way this is cleaner now, for consistency could we move this declaration to the place the other one is?

Reply:
actually no, this event is unique to the player isn't it?

Reply:
Correct this is unique to the player